notebooks/pixtral/gradio_helper.py

```diff
@@ -0,0 +1,126 @@
+                raise gr.Error("You need to upload an image for Llama-3.2-Vision to work. Close the error and try again with an Image.")
+        except NameError:
+            # Handle the case where 'image' is not defined at all
+            raise gr.Error("You need to upload an image for Llama-3.2-Vision to work. Close the error and try again with an Image.")
+
+        conversation = []
+        flag = False
+        for user, assistant in history:
+            if assistant is None:
+                # pass
+                flag = True
+                conversation.extend([{"role": "user", "content": []}])
+                continue
+            if flag == True:
+                conversation[0]["content"] = [{"type": "text", "content": f"{user}"}, {"type": "image"}]
+                conversation.extend([{"role": "assistant", "content": [{"type": "text", "text": assistant}]}])
+                flag = False
+                continue
+            conversation.extend(
+                [{"role": "user", "content": [{"type": "text", "content": user}]}, {"role": "assistant", "content": [{"type": "text", "content": assistant}]}]
+            )
+
+        if len(history) == 0:
+            conversation.append({"role": "user", "content": [{"type": "text", "content": f"{message_text}"}, {"type": "image"}]})
+        else:
+            conversation.append({"role": "user", "content": [{"type": "text", "content": message_text}]})
+        print(f"prompt is -\n{conversation}")
+        prompt = processor.apply_chat_template(conversation, tokenize=False, add_generation_prompt=True)
+        print(f"prompt is -\n{prompt}")
+        image = Image.open(image)
+        image = resize_with_aspect_ratio(image)
+        inputs = processor(prompt, image, return_tensors="pt")
+
+        streamer = TextIteratorStreamer(
+            processor,
+            **{
+                "skip_special_tokens": True,
+                "skip_prompt": True,
+                "clean_up_tokenization_spaces": False,
+            },
+        )
+        generation_kwargs = dict(
+            inputs,
+            streamer=streamer,
+            max_new_tokens=1024,
+            do_sample=False,
+            temperature=0.0,
+            eos_token_id=processor.tokenizer.eos_token_id,
+        )
+
+        thread = Thread(target=model.generate, kwargs=generation_kwargs)
+        thread.start()
+
+        buffer = ""
+        for new_text in streamer:
+            buffer += new_text
+            yield buffer
+
+    demo = gr.ChatInterface(
+        fn=bot_streaming,
+        title=f"{model_name} with OpenVINO",
+        examples=[
+            {"text": "What is the text saying?", "files": ["./small.png"]},
+            {"text": "What does the chart display?", "files": ["./chart.png"]},
+        ],
+        description=f"{model_name} with OpenVINO. Upload an image and start chatting about it, or simply try one of the examples below. If you won't upload an image, you will receive an error.",
```

Comment:
I receive an error `TypeError: can only concatenate list (not "str") to list` if I continue chatting after first message. It doesn't matter what's in the second message, just a message or a message with a picture.

Reply:
fixed